kotlin.html.markdown

```diff
@@ -0,0 +1,401 @@
+---
+
+language: kotlin
+contributors:
+
+filename: LearnKotlin.py
+
+---
+
+Kotlin is a staticaly-typed programming language that runs on the Java Virtual 
+Machine and also can be compiled to JavaScript source code.
+[Read more here.](http://kotlinlang.org/docs/reference/faq.html)
+
+```java
+
+// Single-line comments start with //
+
+/*
+Multi-line comments look like this.
+*/
+
+// Package specification should be at the top of the source file, but need not
+// to match the directory structure.
+package my.demo
+
+// Importing packages from java
+import java.util.ArrayList;
+import java.security.*;
+
+// import foo.Bar // Bar is accessible
+// import bar.Bar as bBar // bBar stands for 'bar.Bar'
+
+// _____________________________________________________________________ BASICS
+
+fun kotlinBasics(name: String) {
+
+  // ---- Literals ----
+  
+  println("Decimal:\t" + 1000
+          + "\nHex 0x100:\t" + 0x100
+          + "\nLong value, denoted with capital L 100L:\t" + 100L
+          + "\nBinary 0b00000100:\t" + 0b00000100
+          + "\nOctal literal not supported!"
+          + "\nFloating-point, Double by default 123.5:\t" + 123.5
+          + "\nFloating-point, Double by default 123.5e10:\t" + 123.5e10
+          + "\nFloat tagged with F 123.5f:\t" + 123.5f
+  )
+
+  // String literal, denoted by double quote. String are iterable
+  for (c in "Hello, world!\n")
+      println(c)
+
+  // Characters are not numbers, the following produces error:
+  // val c: Char = 'c'
+  // c == 1 // ERROR
+
+  // Boolean
+  val b0: Boolean = true
+  val b1: Boolean = false
+  println(
+          "Boolean And (true && false):\t" + (b0 && b1)
+          + "\nBoolean Or (true || false):\t" + (b0 || b1)
+  )
+
+  // Arrays
+  val x: IntArray = intArray(1, 2, 3)
+  // Translates to x.set(0, x.get(1) + x.get(2))
+  x[0] = x[1] + x[2]
+  // arrays have size(), get(), set(), iterator() by default.
+  // [more about array class](http://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-array/index.html)
+  println(
+          "array size():\t" + x.size()
+          + "\nget(0):\t" + x.get(0)
+  )
+
+  // Templates in strings. i.e. pieces of code that are evaluated and whose
+  // results are concatenated into the string.
+  val i = 10
+  val sDemo = "abc"
+  println("i = $i") // evaluates to "i = 10"
+  println("$sDemo.length is ${sDemo.length}") // "abc.length is 3"
+
+
+  // ---- Variables Definition ----
+
+  
+  // Local read-only variable.
+  val a: Int = 1;
+  val b = 1
+  val c: Int // Type required when no initializer is provided
+  c = 1
+  
+  // The following line generates compiler error
+  // a += 1
+
+  // Local mutable value, it's type is inferred.
```

Comment:
If the documentation used "inferred", it's probably better as-is, but it's not what's actually called type inference